src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -33,8 +31,16 @@ private static Task FlushAsync(WritableBuffer writeBuffer)
             var awaitable = writeBuffer.FlushAsync();
             if (awaitable.IsCompleted)
             {
-                awaitable.GetResult();
-                return _completedTask;
+                try
+                {
+                    awaitable.GetResult();
+                }
+                catch(Exception e)
```

Comment:
`awaitable` is a a `WritableBufferAwaitable` and doesn't have an `IsFauted` property